ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue

```diff
@@ -0,0 +1,120 @@
+                  filled
+                  label="Street Address"
+                  :name="Math.random()"
+                  persistent-hint
+                  v-model="address.street"
+                  :rules="[...addressSchema.street]"
+                />
+            </div>
+            <div class="form__row">
+              <v-textarea
+                id="streetAdditional"
+                auto-grow
+                filled
+                class="street-address-additional"
+                label="Additional Street Address (Optional)"
+                :name="Math.random()"
+                rows="1"
+                v-model="address.streetAdditional"
+              />
+            </div>
+            <div class="form__row two-column">
+              <v-row>
+                <v-col>
+                  <v-text-field
+                    id="city"
+                    filled
+                    class="item address-city"
+                    label="City"
+                    :name="Math.random()"
+                    v-model="address.city"
+                    :rules="[...addressSchema.city]"
+                  />
+                </v-col>
+                <v-col>
+                  <v-text-field
+                    id="region"
+                    label="Province"
+                    class="item address-region"
+                    filled
+                    disabled
+                    hint="Address must be in B.C."
+                    persistent-hint
+                    :name="Math.random()"
+                    v-model="address.region"
+                    :rules="[...addressSchema.region]"
+                  />
+                </v-col>
+              </v-row>
+            </div>
+          </v-form>
+
+      </v-col>
+    </v-row>
+  </v-card>
+</template>
+
+<script lang="ts">
+/* eslint-disable no-unused-vars */
+import { defineComponent, reactive, toRefs, watch } from '@vue/composition-api'
+import { CivicAddressSchema } from '@/schemas/civic-address'
+import { useActions } from 'vuex-composition-helpers'
+/* eslint-enable no-unused-vars */
+export default defineComponent({
+  name: 'HomeCivicAddress',
+  components: {},
+  props: {
+    /* used for readonly mode vs edit mode */
+    editing: {
+      type: Boolean,
+      default: false
+    }
+  },
+  setup () {
+    const {
+      setMhrLocation
+    } = useActions<any>([
+      'setMhrLocation'
+    ])
+
+    const addressSchema = CivicAddressSchema
+
+    const localState = reactive({
+      isValidLot: false,
+      showAllAddressErrors: false,
+      address: {
+        street: '',
+        streetAdditional: '',
+        city: '',
+        region: 'British Columbia'
+      }
+    })
+
+    /** Apply local model updates to store. **/
+    watch(() => localState, async () => {
+      // Set civic address data to store
+      for (const [key, value] of Object.entries(localState)) {
+        await setMhrLocation({ key, value })
+      }
+    }, { deep: true })
```

Comment:
Just be aware that that by watching `localState` vs a defined object within your localState you will set anything that changes here to store, which may be unintentional. 
ie if you add a flag that indicates validity of your component here, when it changes from its default state, it will be set to store. 
To that point, it may be safer to watch and use the entries of `localState.address`.

Reply:
To add to this, this setter will work on the root layer of `location` properties where `address` is nested in that object. 
ie `location.address` so it might easier/simpler to create an action/mutation to set location.address.